r2/r2/lib/media.py

```diff
@@ -86,6 +303,36 @@ def str_to_image(s):
     image = Image.open(s)
     return image
 
+def svg_size(image):
+    """calculate the size of an SVG image"""
+    image = BeautifulSoup.BeautifulSoup(cStringIO.StringIO(image))
+    try:
+        dimensions = [image.svg['width'], image.svg['height']]
+
+        # Borrowed and edited from github.com/Zverik/svg-resize
+        for i in (0,1):
+            parts = re.match(r'^\s*(-?\d+(?:\.\d+)?)\s*(px|in|cm|mm|pt|pc|%)?', dimensions[i])
+            num = float(parts.group(1))
+            units = parts.group(2)
+            
+            if units == 'pt':
+                dimensions[i] = num * 1.25
+            elif units == 'pc':
+                dimensions[i] = num * 15.0
+            elif units == 'in':
+                dimensions[i] = num * 90.0
+            elif units == 'mm':
+                dimensions[i] = num * 3.543307
+            elif units == 'cm':
+                dimensions[i] = num * 35.43307
+            elif units == '%':
+                dimensions[i] = -num / 100.0
+            else:
+                dimensions[i] = num
+    except KeyError:
+        dimensions = [500, 500]
```

Comment:
Any particular logic to these defaults? What if `width` is specified and very large, but `height` is unspecified or vice-versa? Can other stuff like a `style` on the root `svg` tag affect the effective dimensions?